File: 20221005/모범답안/20221005_09/MatrixHasOutline.js
/**
 * @param matrix {boolean[][]}
 * @returns {boolean}
 */
function solution(matrix) {
  const n = matrix.length

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if ((i === 0 && j === n - 1) || (j === 0 && i === n -1)) {
        if (!matrix[i][j]) {
          return false
        }
      }
    }
  }

  return true
}

export default solution

File: 20221005/모범답안/20221005_09/MatrixHasOutline.py
def solution(matrix):
    """
    :param matrix: boolean[][]
    :return: boolean
    """

    n = len(matrix)
    for i in range(n):
        for j in range(n):
            if ((i == 0 and j == n - 1) or (j == 0 and i == n - 1)) and not matrix[i][j]:
                return False

    return True
